LoopKit/InsulinKit/SyncPumpEvent.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  SyncPumpEvent.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 12/16/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public struct SyncPumpEvent: Equatable {
```

Comment:
`PersistedPumpEvent` is the type in LoopKit that is designed for representing pump events stored in DoseStore. Why is this being created and returned from queries instead?

Reply:
Primarily because `PersistedPumpEvent` has a number of fields that don't need to be exposed to the remote data services. Plus, `type` is optional while it should really be required for remote data services. It isn't a big deal, but if you feel strongly I can replace it.

Reply:
Exposing a protocol is a nice way to restrict visibility if you really need to hide internals from a public api. You can make PersistedPumpEvent As for `type`, what happens if you encounter a pump event where it is nil? What does that look like when being turned into a `SyncPumpEvent`? It does seem like the sole role of PersistedPumpEvent was exposing pump events to the query interface, so I think now we have two types that users of DoseStore will see that represent PumpEvents stored in DoseStore.

Reply:
Specifically, what do you want me to do here.

Reply:
Regarding your questions about when then pump event type is `nil`, it is explicitly ignored. The fact that the `type` property is optional appears to be an artifact of history. The `type` of a pump event will either be the explicitly specified type or the type of the explicitly specified dose. While it is technically possible from an API perspective to specify `nil` for both, it is not in actual usage (except in some tests). Therefore, it is an "invalid" pump event and, thus, ignored.